tokio/src/sync/semaphore.rs

```diff
@@ -105,6 +150,25 @@ impl Semaphore {
     /// Otherwise, this returns a [`SemaphorePermit`] representing the
     /// acquired permit.
     ///
+    /// # Examples
+    ///
+    ///```
+    /// use tokio::sync::Semaphore;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let semaphore = Semaphore::new(2);
+    ///     let p1 = semaphore.acquire().await;
+    ///     let p2 = semaphore.acquire().await;
+    ///     assert_eq!(p1.is_ok(), true);
+    ///     assert_eq!(p2.is_ok(), true);
+    ///     assert_eq!(semaphore.available_permits(), 0);
+    ///     semaphore.close();
+    ///     let p3 = semaphore.acquire().await;
+    ///     assert_eq!(p3.is_err(), true);
```

Comment:
I think the example could be a bit simpler. Same applies to the `acquire_many` method.
```suggestion
    /// #[tokio::main]
    /// async fn main() {
    ///     let semaphore = Semaphore::new(2);
    ///
    ///     let permit_1 = semaphore.acquire().await.unwrap();
    ///     assert_eq!(semaphore.available_permits(), 1);
    ///
    ///     let permit_2 = semaphore.acquire().await.unwrap();
    ///     assert_eq!(semaphore.available_permits(), 0);
    ///
    ///     drop(permit_1);
    ///     assert_eq!(semaphore.available_permits(), 1);
```